ui/lib/pki/addon/components/page/pki-configuration-edit.ts

```diff
@@ -53,18 +53,36 @@ export default class PkiConfigurationEditComponent extends Component<Args> {
   @waitFor
   *save(event: Event) {
     event.preventDefault();
-    try {
-      for (const model of ['cluster', 'acme', 'urls', 'crl']) {
-        // only call save() if user has permission
-        if (this.args[model as keyof Args].canSet) {
-          yield this.args[model as keyof Args].save();
-          this.flashMessages.success(`Successfully updated ${model} config`);
-        }
-      }
-      this.router.transitionTo('vault.cluster.secrets.backend.pki.configuration.index');
-    } catch (error) {
+    yield this.performSave();
+
+    if (this.errors.length) {
       this.invalidFormAlert = 'There was an error submitting this form.';
-      this.errorBanner = errorMessage(error);
+    } else {
+      this.router.transitionTo('vault.cluster.secrets.backend.pki.configuration.index');
+    }
```

Comment:
Yeah, I hadn't reset `invalidFormAlert` since we transition away on success, but I can update that if it seems important enough. 

I agree - I think `performSave()` can go back into the task here. I'd only pulled it out when the logic seemed to get a little rambly. But it's not too bad

Reply:
Sounds good. I don't think unsetting `invalidFormAlert` is important now that I look at it again. The component will get torn down on transition anyway.